src/pnpq/devices/polarization_controller_base.py

```diff
@@ -0,0 +1,92 @@
+from abc import ABC, abstractmethod
+from typing import TypedDict
+
+import structlog
+from pint import Quantity
+
+from ..apt.connection import AptConnection
+from ..apt.protocol import (
+    AptMessage_MGMSG_MOT_GET_USTATUSUPDATE,
+    ChanIdent,
+    JogDirection,
+)
+
+
+class PolarizationControllerParams(TypedDict):
+    #: Dimensionality must be ([angle] / [time]) or mpc320_velocity
+    velocity: Quantity
+    #: Dimensionality must be [angle] or mpc320_step
+    home_position: Quantity
+    #: Dimensionality must be [angle] or mpc320_step
+    jog_step_1: Quantity
+    #: Dimensionality must be [angle] or mpc320_step
+    jog_step_2: Quantity
+    #: Dimensionality must be [angle] or mpc320_step
+    jog_step_3: Quantity
+
+
+class PolarizationControllerBase(ABC):
+    connection: AptConnection
+
+    log = structlog.get_logger()
+
+    # Setup channels for the device
+    available_channels: frozenset[ChanIdent] = frozenset([])
+
+    def get_status_all(self) -> tuple[AptMessage_MGMSG_MOT_GET_USTATUSUPDATE, ...]:
+        all_status = []
+        for channel in self.available_channels:
+            status = self.get_status(channel)
+            all_status.append(status)
+        return tuple(all_status)
+
```

Comment:
The easiest way to avoid issues is probably to remove all implementation from the ABC and use it purely as an abstract base class.

Consider that the constructor may be different between implementations: our current real-device implementations need an `AptConnection`, but the stub device does not need an `AptConnection`. If we somehow had a third, non-APT device use this interface (unlikely, but maybe possible with refactoring), it would also not need an `AptConnection`.

```suggestion
```